factory-containers/assemble-system-image

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/python3
+#
+# Copyright (c) 2019 Foundries.io
+# SPDX-License-Identifier: Apache-2.0
+#
+import subprocess
+import os
+import requests
+import json
+import argparse
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def get_targets_from_api(factory: str, build_num: str, token: str):
+    api_base_url = 'https://api.foundries.io/ota/repo/'
+    targets_endpoint = 'api/v1/user_repo/targets.json'
+
+    res_targets = []
+    target_url = os.path.join(api_base_url, factory, targets_endpoint)
+    target_resp = requests.get(target_url, headers={'OSF-TOKEN': token})
+    target_resp.raise_for_status()
+
+    for target_name, target in target_resp.json()['signed']['targets'].items():
+        custom = target.get('custom')
+        if not custom:
+            continue
+
+        if custom.get('version', '') == build_num and custom.get('targetFormat', 'NONE') == 'OSTREE':
+            res_targets.append({target_name: target})
+
+    return res_targets
+
+
+# ${IMAGE_BASENAME}-${MACHINE}.wic.gz
+# https://api.foundries.io/projects/msul-dev01/lmp/builds/224/runs/raspberrypi3-64/lmp-factory-image-raspberrypi3-64.wic.gz
+def get_system_image_from_ci(base_uri: str, image_name: str, image_machine: str, token: str):
+    base_uri = base_uri.replace('ci.foundries.io', 'api.foundries.io')
+    target_image_file = '{}-{}.wic.gz'.format(image_name, image_machine)
+    image_url = os.path.join(base_uri, 'runs', image_machine, target_image_file)
+    logger.info('Fetching ' + image_url)
+    image_resp = requests.get(image_url, headers={'OSF-TOKEN': token})
```

Comment:
I think you need "stream=True" here to accomplish what you want with the chunking below: https://requests.readthedocs.io/en/master/user/advanced/#body-content-workflow